optimum/exporters/neuron/__main__.py

```diff
@@ -14,141 +14,230 @@
 # limitations under the License.
 """Entry point to the optimum.exporters.neuron command line."""
 
-import copy
+import argparse
+import inspect
 from argparse import ArgumentParser
+from pathlib import Path
+from typing import Any, Dict, Optional, Union
 
-from ...neuron.utils import is_neuron_available, is_neuronx_available, store_compilation_config
-from ...neuron.utils.version_utils import get_neuroncc_version, get_neuronxcc_version
+from requests.exceptions import ConnectionError as RequestsConnectionError
+
+from ...neuron.utils import is_neuron_available, is_neuronx_available
 from ...utils import logging
 from ...utils.save_utils import maybe_save_preprocessors
 from ..error_utils import AtolError, OutputMatchError, ShapeError
 from ..tasks import TasksManager
-from .convert import export, validate_model_outputs
+from .convert import export_models, validate_models_outputs
 from .model_configs import *  # noqa: F403
+from .utils import (
+    build_stable_diffusion_components_mandatory_shapes,
+    get_stable_diffusion_models_for_export,
+)
 
 
 if is_neuron_available():
     from ...commands.export.neuron import parse_args_neuron
 
     NEURON_COMPILER = "Neuron"
-    NEURON_COMPILER_TYPE = "neuron-cc"
-    NEURON_COMPILER_VERSION = get_neuroncc_version()
+
 
 if is_neuronx_available():
     from ...commands.export.neuronx import parse_args_neuronx as parse_args_neuron  # noqa: F811
 
     NEURON_COMPILER = "Neuronx"
-    NEURON_COMPILER_TYPE = "neuronx-cc"
-    NEURON_COMPILER_VERSION = get_neuronxcc_version()
 
 
 logger = logging.get_logger()
 logger.setLevel(logging.INFO)
 
 
-def main():
-    parser = ArgumentParser(f"Hugging Face Optimum {NEURON_COMPILER} exporter")
-
-    parse_args_neuron(parser)
+def infer_compiler_kwargs(args: argparse.Namespace):
+    # infer compiler kwargs
+    auto_cast = None if args.auto_cast == "none" else args.auto_cast
+    auto_cast_type = None if auto_cast is None else args.auto_cast_type
+    compiler_kwargs = {"auto_cast": auto_cast, "auto_cast_type": auto_cast_type}
+    if hasattr(args, "disable_fast_relayout"):
+        compiler_kwargs["disable_fast_relayout"] = getattr(args, "disable_fast_relayout")
+    if hasattr(args, "disable_fallback"):
+        compiler_kwargs["disable_fallback"] = getattr(args, "disable_fallback")
 
-    # Retrieve CLI arguments
-    args = parser.parse_args()
-    args.output = args.output.joinpath("model.neuron")
+    return compiler_kwargs
 
-    if not args.output.parent.exists():
-        args.output.parent.mkdir(parents=True)
 
-    # Infer the task
-    task = args.task
+def infer_task(task: str, model_name_or_path: str):
```

Comment:
Why? The `TasksManager` already does that no?

Reply:
Yes it does, here I just leverage the function in the `TasksManager` and combine it with checkers. I split it out in  a separate function for code clarity.

Reply:
Would it make sense to add those checks in the `TasksManager`?

Reply:
Those checkers(or at least the message) are kinda CLI specific, so I think it would make sense to just keep them here.